provider/confluence/index.ts

```diff
@@ -0,0 +1,62 @@
+import type {
+    ItemsParams,
+    ItemsResult,
+    MentionsParams,
+    MentionsResult,
+    MetaParams,
+    MetaResult,
+    Provider,
+} from '@openctx/provider'
+import { type Page, listPages } from './api.js'
+
+export type Settings = {
+    host: string
+    port?: string
+    email: string
+    apiToken: string
+}
+
+const confluenceProvider: Provider = {
+    meta(params: MetaParams, settings: Settings): MetaResult {
+        return { name: 'Confluence Pages', mentions: {} }
+    },
+
+    async mentions(params: MentionsParams, settings: Settings): Promise<MentionsResult> {
+        const spaces = await listPages(settings, params.query)
+        return spaces.map(page => {
+            return {
+                title: page.title,
+                uri: page.uri,
+                description: page.space.name,
+                ui: {
+                    hover: {
+                        text: page.title,
+                    },
+                },
+                data: {
+                    page: page,
+                },
```

Comment:
sidenote: @thenamankumar @sqs I know in cody we take care not to send the `content` field over the internal RPC in vscode / we try avoid marshalling it in the transcript. I am wondering if we do the same for data, since for example this can be a large value.

Reply:
Yeah good call. I've just updated it so it only passes the `id` through as data, and the page data is fetched when the items are generated